### PR TITLE
WIFI-1308 Validate EAP parameters before UCI update

### PR DESCRIPTION
**Pull request by RickSommerville**
Signed-off-by: Rick Sommerville <rick.sommerville@netexperience.com>

### CODE REVIEW

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c [hunk 1]
```diff
@@ -302,12 +303,17 @@ static void vif_config_security_set(struct blob_buf *b,
...
-					   SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_IP));
-			blobmsg_add_string(b, "auth_port",
-					   SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_PORT));
-			blobmsg_add_string(b, "auth_secret",
-					   SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_SECRET));
+			auth_server = SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_IP);
+			auth_port   = SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_PORT);
+			auth_secret = SCHEMA_KEY_VAL(vconf->security, SCHEMA_CONSTS_SECURITY_RADIUS_SECRET);
+			LOGT("%s: Server IP %s port %s secret %s", vconf->if_name, auth_server, auth_port, auth_secret);
+			if (!auth_server[0] || !auth_port[0] || !auth_secret[0]) {
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:310]: Aren't all of them mandatory? If yes, &&?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c [hunk 2]
```diff
@@ -1317,11 +1335,11 @@ bool target_vif_config_set2(const struct schema_Wifi_VIF_Config *vconf,
...
 			int num_cconfs)
 {
-
+	int rc;
 	if (!strcmp(vconf->mode, "mesh")) {
-		mesh_vif_config_set(rconf, vconf, changed);
+		rc = mesh_vif_config_set(rconf, vconf, changed);
 	} else {
-		ap_vif_config_set(rconf, vconf, changed);
+		rc = ap_vif_config_set(rconf, vconf, changed);
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1342]: Return is not checked. Should we check and log failure case?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 1 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:310]: The accounting entries are not mandatory.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 2 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1342]: I generated a log at the point of detection.  Passed the error up the chain, but there is nothing useful to be done beyond this function.

**Review by ArifAlam — APPROVED** [review]

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 1 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:310]: The first byte of each string will be "/0" if it is empty.  This evaluate to zero so the ! causes 1 and the error will be triggered if any string is empty.

**Review by yashvardhan-netex — APPROVED** [review]

**Comment by RickSommerville** [comment]
Merged

**Pull request closed without merging**